src/shell/shell_cmds.cpp

```diff
@@ -465,26 +472,250 @@ static std::string to_search_pattern(const char *arg)
 	char buffer[CROSS_LEN];
 	pattern = ExpandDot(pattern.c_str(), buffer, sizeof(buffer));
 
-	// When there's no wildcard and target is a directory then search files
-	// inside the directory.
-	const char *p = pattern.c_str();
-	if (!strrchr(p, '*') && !strrchr(p, '?')) {
-		uint16_t attr = 0;
-		if (DOS_GetFileAttr(p, &attr) && (attr & DOS_ATTR_DIRECTORY))
-			pattern += "\\*.*";
+	return pattern;
+}
+
+uint32_t byte_count,file_count,dir_count;
+Bitu p_count;
+std::vector<std::string> dirs, adirs;
+
+static size_t GetPauseCount() {
+	return (real_readb(BIOSMEM_SEG,BIOSMEM_NB_ROWS) > 2u) ? (real_readb(BIOSMEM_SEG,BIOSMEM_NB_ROWS)) : 24u;
+}
+
+static bool dirPaused(DOS_Shell * shell, Bitu w_size, bool optP, bool optW) {
+	p_count+=optW?5:1;
+	if (optP && p_count%(GetPauseCount()*w_size)<1) {
+		shell->WriteOut(MSG_Get("SHELL_CMD_PAUSE"));
+		Bit8u c;Bit16u n=1;
+		DOS_ReadFile(STDIN,&c,&n);
+		if (c==3) {shell->WriteOut("^C\r\n");return false;}
+		if (c==0) DOS_ReadFile(STDIN,&c,&n); // read extended key
```

Comment:
There are a whole 'lotta magic numbers going on here.  Suggest comments or, better, if the code can be made "self commenting" though intermediate const variables (with nice names), that would be idea.